### DISCUSSION TITLE
How to access vscode built-in extension API from theia extension?

### DISCUSSION THREAD

**Discussion by thers** [eclipse-theia/theia#8657] [Q&A] [answered]
Some vscode built-in extensions expose their API, how can I access such from theia extension?

For example, `typescript-language-features` expose `configurePlugin` method: https://github.com/microsoft/vscode/blob/ff915844119ce9485abfe8aa9076ec76b5300ddd/extensions/typescript-language-features/src/api.ts#L15

**Comment by vince-fugnitto** [comment, marked as answer]
@thers the question has been previously asked: https://github.com/eclipse-theia/theia/discussions/8126.
It should be possible to perform this communication between plugins however using [configurePlugin](https://code.visualstudio.com/api/references/contribution-points#Plugin-configuration) as you mentioned.

**Reply by thers** [reply]
So if im getting it right: I'd need to make vscode extension for that, is that correct?

**Reply by vince-fugnitto** [reply]
> So if im getting it right: I'd need to make vscode extension for that, is that correct?

Yes that'd be the best option if you'd like to use the exposed API from another plugin.
